core/src/test/java/io/grpc/internal/ServerImplTest.java

```diff
@@ -1030,6 +1032,28 @@ public void cancelled(Context context) {
     return streamListener;
   }
 
+  @Test
+  public void testContextExpiredBeforeStreamCreate_StreamCancelNotCalledBeforeSetListener()
+      throws Exception {
+    AtomicBoolean contextCancelled = new AtomicBoolean(false);
+    AtomicReference<Context> context = new AtomicReference<>();
+    AtomicReference<ServerCall<String, Integer>> callReference
+        = new AtomicReference<>();
+
+    testStreamClose_setup(callReference,
+        context, contextCancelled, 0L);
+
+    // This assert that stream.setListener() is called before stream.cancel(), which avoid every
+    // short deadlines causing NPEs.
+    InOrder inOrder = inOrder(stream);
+    inOrder.verify(stream).setListener(any(ServerStreamListener.class));
+    inOrder.verify(stream).cancel(statusCaptor.capture());
+
+    assertThat(
+        statusCaptor.getValue().asException()).hasMessageThat().contains("context timed out");
```

Comment:
Please wrap before the hasMessageThat(), since then it keeps the argument to assertThat together.